plugins/hypervisors/xenserver/src/com/cloud/hypervisor/XenServerGuru.java

```diff
@@ -185,7 +185,8 @@ public boolean trackVmHostChange() {
             DataTO srcData = cpyCommand.getSrcTO();
             DataTO destData = cpyCommand.getDestTO();
 
-            if (srcData.getObjectType() == DataObjectType.SNAPSHOT && destData.getObjectType() == DataObjectType.TEMPLATE) {
+            if (srcData.getHypervisorType() == HypervisorType.XenServer && srcData.getObjectType() == DataObjectType.SNAPSHOT &&
```

Comment:
@anshul1886  I looked it up. @bhaisaab @ustcweizhou Anshul is right about the equals versus ==, for readability it makes sense to leave this as is. from 8.9 in the language specs:

Because there is only one instance of each enum constant, it is permitted to use the == operator in place of the equals method when comparing two object references if it is known that at least one of them refers to an enum constant.